### PR TITLE
feat: add --output markdown format

### PR DESCRIPTION
**Pull request by nieblara**
## Summary

- Registers `markdown` as a valid `--output` kind alongside `json` and `plaintext`
- Adds markdown rendering for resource output: GFM tables for lists, key-value bullets for singular resources, and a rich flag view with environment status table
- Extracts `paginationSuffix` helper to share between plaintext and markdown paths

> **Stacked on** `feat/dry-run-flag`

## Test plan

- [x] All existing `internal/output` tests pass
- [x] New unit tests for markdown rendering helpers (`markdown_test.go`)
- [x] Integration tests through `CmdOutput` for singular, list, pagination, empty, and unknown resource cases (`resource_output_test.go`)
- [x] Command-level tests for `archive`, `toggle-on`, `toggle-off` with `--output markdown`

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Medium Risk**
> Adds a new output mode that changes user-visible formatting paths across many commands; risk is mainly in rendering/edge cases (pagination, empty lists) rather than data or auth logic.
> 
> **Overview**
> Adds `markdown` as a supported `--output` kind (alongside `json`/`plaintext`) and updates CLI/config help + validation error messages accordingly.
> 
> Implements GitHub-flavored markdown rendering for resource output: tables for lists, bullet key/value blocks for singular resources, and a richer flag view with an environments status table and extra metadata. Refactors pagination formatting into a shared `paginationSuffix` helper and extends command/unit tests to cover markdown output and edge cases (empty results, unknown resources, pagination, `--fields` ignored outside JSON).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 36e05762e2003d8f170eb5b8964ae77faae8cfac. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by nieblara** [comment]
@copilot resolve the merge conflicts in this pull request

**Review by ari-launchdarkly — COMMENTED** [review]
Did we add markdown as an output? We should test that too if we can. It looks like it was added to the description but I'm not sure if that's just a change to align it to reality

**Comment by nieblara** [comment]
> Did we add markdown as an output? We should test that too if we can. It looks like it was added to the description but I'm not sure if that's just a change to align it to reality

ah @ari-launchdarkly seems like we had a few conflicts that i needed to resolve before the updated diff showed up for you - so i think its good for you to check out again! the changes here should include the markdown output

**Comment by ari-launchdarkly** [comment]
@copilot would you mind reviewing this PR?

File: internal/output/markdown.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+		headers[i] = col.Header
+		separators[i] = "---"
+	}
+
+	var sb strings.Builder
+	sb.WriteString("| ")
+	sb.WriteString(strings.Join(headers, " | "))
+	sb.WriteString(" |\n| ")
+	sb.WriteString(strings.Join(separators, " | "))
+	sb.WriteString(" |")
```
Comment (ari-launchdarkly) [inline internal/output/markdown.go:23]: is there any value to us having a utility to build this out? Just curious, not asking for a change

File: internal/output/markdown.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+	for _, item := range items {
+		vals := make([]string, len(cols))
+		for i, col := range cols {
+			vals[i] = escapeMDPipe(colValue(item, col))
```
Comment (ari-launchdarkly) [inline internal/output/markdown.go:28]: do we need error handling here?

**Review by ari-launchdarkly — APPROVED** [review]
Great work!

File: internal/output/markdown.go (hunk 1 above)
Comment (nieblara) [reply internal/output/markdown.go:23]: right now, just dont see the need for a dependency for something im not sure will change often yet - but ill keep an eye out for something suitable!

File: internal/output/markdown.go (hunk 2 above)
Comment (nieblara) [reply internal/output/markdown.go:28]: a good idea - deferring to a future pr
